README-template.md

```diff
@@ -0,0 +1,1140 @@
+  creation permission. `materializationProject` is optional.
+
+**Note:** As mentioned in the [BigQuery documentation](https://cloud.google.com/bigquery/docs/writing-results#temporary_and_permanent_tables),
+the queried tables must be in the same location as the `materializationDataset`.
+Also, if the tables in the `SQL statement` are from projects other than the
+`parentProject` then use the fully qualified table name i.e.
+`[project].[dataset].[table]`.
+
+**Important:** This feature is implemented by running the query on BigQuery and
+saving the result into a temporary table, of which Spark will read the results
+from. This may add additional costs on your BigQuery account.
+
+### Reading From Views
+
+The connector has a preliminary support for reading from
+[BigQuery views](https://cloud.google.com/bigquery/docs/views-intro). Please
+note there are a few caveats:
+
+* BigQuery views are not materialized by default, which means that the connector
+  needs to materialize them before it can read them. This process affects the
+  read performance, even before running any `collect()` or `count()` action.
+* The materialization process can also incur additional costs to your BigQuery
+  bill.
+* By default, the materialized views are created in the same project and
+  dataset. Those can be configured by the optional `materializationProject`
+  and `materializationDataset` options, respectively. These options can also
+  be globally set by calling `spark.conf.set(...)` before reading the views.
+* Reading from views is **disabled** by default. In order to enable it,
+  either set the viewsEnabled option when reading the specific view
+  (`.option("viewsEnabled", "true")`) or set it globally by calling
+  `spark.conf.set("viewsEnabled", "true")`.
+* As mentioned in the [BigQuery documentation](https://cloud.google.com/bigquery/docs/writing-results#temporary_and_permanent_tables),
+  the `materializationDataset` should be in same location as the view.
+
+### Writing data to BigQuery
+
+#### Direct write using the BigQuery Storage Write API
+The Spark 2.4 dedicated connector supports writing directly to BigQuery without first writing to GCS, using
+the [BigQuery Storage Write API](https://cloud.google.com/bigquery/docs/write-api)
+to write data directly to BigQuery. In order to enable this option, please set the `writeMethod` option
+to `direct`, as shown below:
+
+```
+df.write \
+  .format("bigquery") \
+  .option("writeMethod", "direct") \
+  .save("dataset.table")
+```
+
+Partition overwriting and the use of `datePartition`, `partitionField` and `partitionType` as described below is not
+supported at this moment by the direct write method.
+
+**Important:** Please refer to the [data ingestion pricing](https://cloud.google.com/bigquery/pricing#data_ingestion_pricing)
+page regarding the BigQuery Storage Write API pricing.
+
+**Important:** Please use version ${next-release-tag} and above for direct writes, as previous
```

Comment:
Here the version should be the constant 0.24.2, due to the PR #563 fix

Reply:
Updated line 252 to use the constant instead of the varible.